oio/common/client.py

```diff
@@ -92,9 +93,15 @@ def _direct_request(self, method, url, headers=None, request_attempts=None,
         if kwargs.get("autocreate"):
             if not headers:
                 headers = dict()
-            headers["X-oio-action-mode"] = "autocreate"
+            headers[HEADER_PREFIX + "action-mode"] = "autocreate"
             kwargs = kwargs.copy()
             kwargs.pop("autocreate")
+        if kwargs.get("tls"):
+            if not headers:
+                headers = dict()
+            kwargs = kwargs.copy()
```

Comment:
I'm not sure this copy is necessary.

Reply:
This copy is unnecessary if the `autocreate` is set but still necessary for other case.
I've updated to avoid the unnecessary copy

Reply:
The copy is not necessary because the kwargs dictionary is guaranteed to be new. When we pop elements out of it, the dictionary seen by the calling method will still see them.
https://stackoverflow.com/questions/45883655/is-it-always-safe-to-modify-the-kwargs-dictionary